### DISCUSSION TITLE
Issue with loading .ttf font in vips_text

### DISCUSSION THREAD

**Discussion by RussDragon** [libvips/libvips#3470] [Q&A]
Hello! Not sure that this is a correct place for reporting such case, but still didn't found any related information to my problem.

**Environment**:
- Strictly jailed from outer filesystem (in linux terms chdir'ed & chroot'ed) CGO + LibVips (C) code. 
- Proprietary font in .ttf format
- Arbitrary text on input
- Vips 8.14.1

**What I need to do**: 
I want to create a text image using vips_text in specified font that should be loaded from file.

**What I tried to do**: 
First of all, I tried to setup some working solution in a basic unjailed env on my own Mac OS system. I've wrote some CGO bindings for vips_text and some aux functions like vips_image_write_to_buffer. 
My calling code looks similar to this: 
`C.vipsText(&outimage, C.CString("arbitrary string"), C.CString("myfont.ttf"), C.CString("My Font Family Name"), imageX, imageY)`

It works nice if I don't use a fontfile and choose some default font (Arial or Helvetica for example) or **if I preinstall "myfont" at first.**

But seems LibVips does not support font usage from a file without preinstalling it first. As far as I could figure out, it because of a "FontConfig" utility and somehow connected to Pango library. And this is also a reason why we can't load font from a memory (am I right?).

It is not a big problem if code works in some fully-operational operation system and I can preinstall font that I need to use. But because of a security reasons, I need to run LibVips (C version of it) in strictly jailed environment when almost everything cut off. 

I can copy font that I need to this jail, and LibVips seems to load it fine, but then I have an error that "FontConfig not found, LastResort not found, using built-in copy".

**So, three main questions**: 
1. Is the task I want to do doable at all in specified conditions? 
2. Am I right that FontConfig and LastResort are some kind of fallback when LibVips can not find Font Family in font file? 
3. If 2 isn't true and LibVips/Pango always use FontConfig to render fonts even if it loaded from file, do you have any ideas how can I accomplish my task in strictly jailed condition? Seems that I need to symlink/copy some configs to jail too? 

I tried to google the third point but haven't found any meaningful information. 

Thank you in advance!

P.S. I also have a problem that LibVips don't recognize my font family even if I preinstall it on my Mac OS system, but seems it is not such important as problem above.

**Comment by jcupitt** [comment]
Hi @RussDragonm,

It should work with uninstalled fonts, or it does on linux at least. I can run:

```
$ vips text x.png "Hello!" --dpi 300  --font "Miss Fajardose"
```

To make:

![x](https://user-images.githubusercontent.com/580843/225765660-eb50c0d8-2403-43f0-b93b-e6771dd7094f.png)

But if I use `--fontfile` to load the uninstalled font like this:

```
$ vips text x.png "Hello!" --dpi 300 --fontfile MissFajardose-Regular.ttf --font "Miss Fajardose"
```

I get:

![x](https://user-images.githubusercontent.com/580843/225765914-a5eb84c4-ec68-4511-b7a2-cc81af0eded0.png)

You need to get the font name exactly correct, could that be the problem? 

macOS uses a different font backend, I don't know if that could be a factor.
